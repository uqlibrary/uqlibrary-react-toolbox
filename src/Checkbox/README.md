# SelectField

A simple Checkbox component that lists items for UQ Library applications using ReactJs

## Props
- `standard props from http://www.material-ui.com/#/components/checkbox`
- `helpTitle: PropTypes.string`
- `helpText: PropTypes.any`


## Usage

**./App.js**
```jsx
import React from 'react';
...

import {Checkbox} from 'uqlibrary-react-toolbox';
 
const App = () => (
  <div>
    <Checkbox
       label="Simple"
     />
  </div>
);

ReactDOM.render(
  <App />,
  document.getElementById('app')
);
```
