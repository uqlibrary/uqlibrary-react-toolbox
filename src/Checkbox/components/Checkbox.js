import React from 'react';
import PropTypes from 'prop-types';

import Checkbox from 'material-ui/Checkbox';
import propFilter from '../../helpers/_filterProps';
import {HelpIcon} from '../../HelpDrawer';

const CheckboxWrapper = props => {
    const filteredProps = propFilter(props, Checkbox.propTypes);
    filteredProps.onCheck = (event, isInputChecked) => props.input.onChange(isInputChecked);
    delete filteredProps.errorText;

    return (
        <div style={{position: 'relative', width: '100%'}} className={props.meta && props.meta.error ? 'error-checkbox' : {}}>
            <Checkbox {...filteredProps} />
            {props.help && props.help.text && (
                <HelpIcon {...props.help} />
            )}
        </div>
    );
};

CheckboxWrapper.propTypes = {
    ...Checkbox.propTypes,
    help: PropTypes.shape({
        title: PropTypes.string,
        text: PropTypes.object,
        buttonLabel: PropTypes.string
    })
};

export default CheckboxWrapper;
