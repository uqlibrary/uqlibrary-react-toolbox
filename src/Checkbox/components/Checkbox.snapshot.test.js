jest.dontMock('./Checkbox');

import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import Checkbox from './Checkbox';

import filterProps from '../../helpers/_filterProps';

function setup(props) {
    const consolidatedProps = filterProps(props, Checkbox.propTypes);
    return shallow(<Checkbox {...consolidatedProps} />);
}

describe('Checkbox snapshots tests', () => {
    it('renders Checkbox component', () => {
        const props = { label: 'This is a test checkbox component'};
        const wrapper = setup(props);
        const tree = toJson(wrapper);
        expect(tree).toMatchSnapshot();
    });
});
