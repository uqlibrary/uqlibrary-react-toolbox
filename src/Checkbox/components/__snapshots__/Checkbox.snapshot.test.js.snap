// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox snapshots tests renders Checkbox component 1`] = `
<div
  className={Object {}}
  style={
    Object {
      "position": "relative",
      "width": "100%",
    }
  }
>
  <Checkbox
    disabled={false}
    label="This is a test checkbox component"
    labelPosition="right"
    onCheck={[Function]}
  />
</div>
`;
